mission_control/navigator_missions/nav_missions/detect_deliver.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+import txros
+import rospy
+import std_srvs.srv
+from navigator_msgs.msg import DockShape
+import numpy as np
+from geometry_msgs.msg import Point
+import sensor_msgs.msg
+import nav_msgs.msg
+from navigator_tools import rosmsg_to_numpy
+import cv2
+import time
+from twisted.internet import defer
+
+error_threshold = 0.1  # how small center error can be to be considered centered
+# what proportion of the image width should be taken by the symbol
+width_proportion = 0.30
+# how small width error can be to be considered correct distance
+width_error_threshold = 0.2
+
+def bounding_rect(points):
+    maxX = 0
+    minX = 2000
+    maxY = 0
+    minY = 2000
+    for i in range(len(points)):
+        if(points[i].x > maxX):
+            maxX = points[i].x
+        if(points[i].y > maxY):
+            maxY = points[i].y
+        if(points[i].x < minX):
+            minX = points[i].x
+        if(points[i].y < minY):
+            minY = points[i].y
+    return np.array([maxX, maxY, minX, minY])
+
+
+def area_of_rect(rect):
+    return (rect[1] - rect[2]) * (rect[1] - rect[3])
+
+
+@txros.util.cancellableInlineCallbacks
+def main(navigator):
+  shooterLoad = rospy.ServiceProxy("/shooter/load", std_srvs.srv.Trigger)
```

Comment:
Your really shouldn't be mixing rospy and txros code. The equivalent in txros is:

    shooterLoad = nh.get_service_client("/shooter/load", std_srvs.srv.Trigger)
    ...
    yield shooterLoad(std_srvs.srv.TriggerRequest())

If this actually works, it means something else is calling `rospy.init()`, which is its own problem. If you use both rospy and txros, your node will appear as two different nodes, which is probably undesirable.